opensrp-chw-core/src/test/java/org/smartregister/chw/core/fragment/CoreBirthNotificationFragmentTest.java

```diff
@@ -0,0 +1,328 @@
+        ReflectionHelpers.setField(coreFpRegisterFragment, "syncProgressBar", syncProgressBar);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "syncButton", imageView);
+        ReflectionHelpers.setField(coreFpRegisterFragment, "globalQrSearch", false);
+
+        CoreLibrary.init(context);
+        when(context.commonrepository(anyString())).thenReturn(commonRepository);
+        activityController = Robolectric.buildActivity(AppCompatActivity.class).create().resume();
+        activity = activityController.get();
+        Context.bindtypes = new ArrayList<>();
+        Whitebox.setInternalState(coreFpRegisterFragment, "clientsView", clientsView);
+        Whitebox.setInternalState(coreFpRegisterFragment, "presenter", presenter);
+        SyncStatusBroadcastReceiver.init(activity);
+    }
+
+    @Test
+    public void testSetupViews() {
+        when(coreFpRegisterFragment.getActivity()).thenReturn(activity);
+        when(coreFpRegisterFragment.getContext()).thenReturn(activity);
+        View view = LayoutInflater.from(activity).inflate(R.layout.fragment_base_register, null);
+        coreFpRegisterFragment.setupViews(view);
+
+        View dueOnlyLayout = view.findViewById(R.id.due_only_layout);
+        dueOnlyLayout.setVisibility(View.VISIBLE);
+        View searchBarLayout = view.findViewById(org.smartregister.R.id.search_bar_layout);
+        searchBarLayout.setBackgroundResource(R.color.customAppThemeBlue);
+        assertEquals(View.VISIBLE, dueOnlyLayout.getVisibility());
+    }
+
+    @Test
+    public void testSetUniqueID() {
+        if (coreFpRegisterFragment.getSearchView() != null) {
+            coreFpRegisterFragment.getSearchView().setText(Mockito.anyString());
+        }
+        assertNotNull(coreFpRegisterFragment.getSearchView());
+    }
+
+    @Test
+    public void getMainConditionCallsPresenterGetCondition() {
+        coreFpRegisterFragment.getMainCondition();
+        verify(coreFpRegisterFragment.presenter(), Mockito.times(1)).getMainCondition();
+    }
+
+    @Test
+    public void testInitializeAdapter() {
+        when(coreFpRegisterFragment.getActivity()).thenReturn(activity);
+        coreFpRegisterFragment.initializeAdapter(new HashSet<>());
+        verify(clientsView).setAdapter(adapterArgumentCaptor.capture());
+        assertNotNull(adapterArgumentCaptor.getValue());
+        assertEquals(20, adapterArgumentCaptor.getValue().currentlimit);
+    }
+
+    @Test
+    public void whenToggleFilterSelectionAnswered() {
+        Mockito.doNothing().when(coreFpRegisterFragment).toggleFilterSelection(view);
+        coreFpRegisterFragment.toggleFilterSelection(view);
+
+        ArgumentCaptor<View> captor = ArgumentCaptor.forClass(View.class);
+        verify(coreFpRegisterFragment, Mockito.times(1)).toggleFilterSelection(captor.capture());
+        assertEquals(captor.getValue(), view);
+    }
+
+    @Test
+    public void testOnViewClick() {
```

Comment:
Kindly follow the test naming convention `test[MethodName][ExpectedResult][Parameters]`